File: engine/model/src/main/scala/net/server/channel/packet/movement/MoveDragonPacket.scala
package net.server.channel.packet.movement

import java.awt.Point

import net.server.MovementData

class MoveDragonPacket(private var _startPosition: Point, private var _hasMovement: Boolean, private var _movementDataList: java.util.List[MovementData], private var _movementList: java.util.List[java.lang.Byte]) extends BaseMovementPacket {
  def startPosition: Point = _startPosition

  def hasMovement: Boolean = _hasMovement

  def movementDataList: java.util.List[MovementData] = _movementDataList

  def movementList: java.util.List[java.lang.Byte] = _movementList
}

File: engine/engine-model/src/main/java/net/server/channel/packet/movement/MoveDragonPacket.java
package net.server.channel.packet.movement;

import java.awt.Point;
import java.util.List;

import net.server.MovementData;

public class MoveDragonPacket extends BaseMovementPacket {
   private final Point startPosition;

   private final Boolean hasMovement;

   private final List<MovementData> movementDataList;

   private final List<Byte> movementList;

   public MoveDragonPacket(Point startPosition, Boolean hasMovement, List<MovementData> movementDataList, List<Byte> movementList) {
      this.startPosition = startPosition;
      this.hasMovement = hasMovement;
      this.movementDataList = movementDataList;
      this.movementList = movementList;
   }

   public Point startPosition() {
      return startPosition;
   }

   public Boolean hasMovement() {
      return hasMovement;
   }

   public List<MovementData> movementDataList() {
      return movementDataList;
   }

   public List<Byte> movementList() {
      return movementList;
   }
}
